Zend/zend_vm_execute.h

```diff
@@ -48790,7 +48790,7 @@ static ZEND_OPCODE_HANDLER_RET ZEND_FASTCALL ZEND_CLONE_SPEC_TMPVAR_HANDLER(ZEND
 			/* Ensure that if we're calling a private function, we're allowed to do so.
 			 */
 			scope = EX(func)->op_array.scope;
-			if (UNEXPECTED(ce != scope)) {
+                if (UNEXPECTED(clone->common.scope != scope)) {
```

Comment:
zend_vm_execute.h is a generated file. The change should be made in zend_vm_def.h and then zend_vm_execute.h can be regenerated using zend_vm_gen.php.

Reply:
I believe this will also not work if the extending class also defines a private `__clone()`. If it doesn't, please add a test and use the `zend_check_private` function.